test/e2e/playbook/provision.yaml

```diff
@@ -119,27 +124,48 @@
           - tomli_w
         state: present
 
-    # We need quite recent containerd that has NRI support so use the one
-    # that is compiled from sources.
-    - name: copy containerd sources
-      copy: src="{{ item }}" dest="/usr/bin" mode=0755
+    # Install containerd from a release tarball to /usr/local.
+    - name: Install containerd binaries from release tarball
+      ansible.builtin.unarchive:
+        src: "https://github.com/containerd/containerd/releases/download/v{{ containerd_release }}/containerd-{{ containerd_release }}-linux-amd64.tar.gz"
+        dest: /usr/local
+        remote_src: yes
+      when: is_containerd and containerd_src == ""
+
+    # Install containerd binaries from compiled sources.
+    - name: Install containerd binaries from compiled sources
+      copy: src="{{ item }}" dest="/usr/local/bin" mode=0755
       with_items:
         - "{{ containerd_src }}/bin/ctr"
         - "{{ containerd_src }}/bin/containerd"
         - "{{ containerd_src }}/bin/containerd-shim"
         - "{{ containerd_src }}/bin/containerd-shim-runc-v1"
         - "{{ containerd_src }}/bin/containerd-shim-runc-v2"
-      when: is_containerd
-
-    # We need quite recent cri-o that has NRI support so use the one
-    # that is compiled from sources.
-    - name: copy cri-o sources
-      copy: src="{{ item }}" dest="/usr/bin" mode=0755
+      when: is_containerd and containerd_src != ""
+
+    # Install CRI-O from a release tarball to /usr/local.
+    - block:
+      - name: Fetch and extract CRI-O release tarball
+        ansible.builtin.unarchive:
+          src: "https://storage.googleapis.com/cri-o/artifacts/cri-o.amd64.v{{ \
+crio_release }}.tar.gz"
```

Comment:
can we maybe have a variable to store the download path and use the variable here instead?

Reply:
Added dedicated `{containerd,crio}_tarball` variables for both containerd and crio release tarball URLs.